backend/graph-proxy/src/graphql/mod.rs

```diff
@@ -50,12 +51,61 @@ pub fn root_schema_builder() -> SchemaBuilder<Query, Mutation, Subscription> {
 
 /// The root query of the service
 #[derive(Debug, Clone, Default, MergedObject)]
-pub struct Query(WorkflowsQuery, WorkflowTemplatesQuery);
+pub struct Query(NodeQuery, WorkflowsQuery, WorkflowTemplatesQuery);
+
+/// Provides Relay node resolver for fetching any Node by ID.
+#[derive(Debug, Clone, Default)]
+pub struct NodeQuery;
+
+#[Object]
+impl NodeQuery {
+    async fn node(&self, ctx: &Context<'_>, id: ID) -> Option<NodeValue> {
```

Comment:
If I understand correctly the goal is to implement GraphQL's Global Object Identification spec (https://graphql.org/learn/global-object-identification/)

This will add something to the graph QL schema that looks like:

type Query {
    node(id: ID): NodeValue
}

How will this interact in the federated graph?

What is other services implement their own node query following this spec? Is the router smart enough to figure out what to do?